tuf/ngclient/fetcher.py

```diff
@@ -40,3 +48,51 @@ def fetch(self, url: str, required_length: int) -> Iterator[bytes]:
             A bytes iterator
         """
         raise NotImplementedError  # pragma: no cover
+
+    @contextmanager
+    def download_file(self, url: str, required_length: int) -> Iterator[IO]:
```

Comment:
required_length is actually more like max_length, right?

Reply:
yes, it is the upper boundary.